clear_cell_outputs/src/index.ts

```diff
@@ -0,0 +1,42 @@
+
+
+
+import { ToolbarButton } from "@jupyterlab/apputils";
+import { DocumentRegistry } from "@jupyterlab/docregistry";
+import { INotebookModel, NotebookPanel } from "@jupyterlab/notebook";
+import { IDisposable } from "@lumino/disposable";
+
+export class ButtonExtension implements DocumentRegistry.IWidgetExtension<NotebookPanel, INotebookModel> {
+
+  createNew(panel: NotebookPanel, context: DocumentRegistry.IContext<INotebookModel>): IDisposable {
+    // Create the toolbar button
+    let clearButton = new ToolbarButton({
+        label: 'Clear all outputs',
+        onClick: () => alert('Cleared all outputs!')
+    });
```

Comment:
To trigger a command, I advice you to switch to a `CommandToolbarButton` (you can imported from `'@jupyterlab/apputils'` too).

```suggestion
    let clearButton = new CommandToolbarButton({
                commands: this.commands, // CommandRegistry
                id: command_id // Command id (this is a string)
      })
```

You need to pass the command registry in the constructor of `ButtonExtension`. So you will be able to access it in the `createNew` method.

Note:

The id of the command to clear all outputs can be found in the jupyterlab code:

https://github.com/jupyterlab/jupyterlab/blob/46e2b9bc4659c0529d32678004fc59ec3e39f0e6/packages/notebook-extension/src/index.ts#L127

Reply:
> * Add the new folder in the top README
> * Add the new folder to the list in the CI
>   https://github.com/jupyterlab/extension-examples/blob/ece34adfbe458351fee356a4a8d4dbc862948793/.github/workflows/main.yml#L15
> * A `Untitled.ipynb` file slip in the PR - to be removed
> * Use `yarn` or `jlpm` not npm - and remove the package-lock.json
> * Remove the license file inside the new example folder
> * Add a screenshot or a GIF

I had done the above changes. But while doing changes in the `clearButton` function I got some errors while running ` jlpm build`. That's why it's taking some time to resolve it.

Reply:
No pressure :wink:  - thanks for the update

You can chat with the community on https://gitter.im/jupyterlab/jupyterlab

For instance, if you want some help with the error you are getting.

Reply:
Hello @fcollonval, I have done the changes you told me to do & gave the button functionality to clear the output of all cells, but I don't use `CommandToolbarButton` because I am facing some errors after using it. But I gave the button functionality as you told me do I really need to use it?

Reply:
This is fine. Could you push your latest changes please?